spec/api-app-spec.js

```diff
@@ -226,6 +226,27 @@ describe('app module', () => {
         })
       })
     })
+
+    it('passes arguments to the second-instance event', async () => {
+      const appPath = path.join(__dirname, 'fixtures', 'api', 'singleton')
+      const first = ChildProcess.spawn(remote.process.execPath, [appPath])
+      const firstExited = emittedOnce(first, 'exit')
+
+      // Wait for the first app to boot.
+      const data1 = await emittedOnce(first.stdout, 'data')
```

Comment:
This doesn't work on Windows because there is no guaruntee that "data" will contain the entire line, in fact it almost certainly won't 😢 

You'll need to buffer all `data` events and split / read the string somehow to determine when you can launch the second app and what JSON you need to parse 👍 

(Check out the windows failures)

Reply:
Ah, I was just copying this from the test above which does the same thing. I'll robustify this.